tests/sdk/v2/logs/test_regrouping.py

```diff
@@ -0,0 +1,137 @@
+################################################################################
+# © Copyright 2023 Zapata Computing Inc.
+################################################################################
+"""
+Unit tests for ``orquestra.sdk._base._logs._regrouping``.
+"""
+
+from pathlib import Path
+
+import pytest
+
+from orquestra.sdk._base._logs import _regrouping
+
+
+class TestIsWorker:
+    @staticmethod
+    @pytest.mark.parametrize(
+        "path",
+        [
+            pytest.param(
+                Path(
+                    "/tmp/ray/session_latest/logs/worker-"
+                    "903b5bfa866935b17bbd3d3a64d22c7c6b746c2d522ad5fca9cf1f6a-"
+                    "01000000-249.err"
+                ),
+                id="ce-path-latest",
+            ),
+            pytest.param(
+                Path(
+                    "session_latest/logs/worker-"
+                    "903b5bfa866935b17bbd3d3a64d22c7c6b746c2d522ad5fca9cf1f6a-"
+                    "01000000-249.err"
+                ),
+                id="ray-tmp-relative-latest",
+            ),
+            pytest.param(
+                Path(
+                    "session_2023-06-12_14-46-40_833957_38917/logs/worker-"
+                    "903b5bfa866935b17bbd3d3a64d22c7c6b746c2d522ad5fca9cf1f6a-"
+                    "01000000-249.err"
+                ),
+                id="ray-tmp-relative-timestamp",
+            ),
+            pytest.param(
+                Path(
+                    "session_latest/logs/worker-"
+                    "903b5bfa866935b17bbd3d3a64d22c7c6b746c2d522ad5fca9cf1f6a-"
+                    "01000000-249.err"
+                ),
+                id="ray-tmp-relative-latest",
+            ),
+            pytest.param(
+                Path(
+                    "/Users/emiliano/.orquestra/ray/session_latest/logs/worker-"
+                    "903b5bfa866935b17bbd3d3a64d22c7c6b746c2d522ad5fca9cf1f6a-"
+                    "01000000-249.err"
+                ),
+                id="local-absolute",
+            ),
+        ],
+    )
+    def test_valid(path: Path):
```

Comment:
Suggestion: You mentioned the risk of Windows paths. Could you consider adding some positive and negative Windows-like paths to this UT?

Reply:
I did look into Windows-specific paths in our test cases. It turns out `Path("/Users/emiliano/.orquestra/ray/session_latest/logs/")` produces `PosixPath` paths when running on UNIX-like system and `WindowsPath` when running on Windows. 

Using `WindowsPath()` directly on macOS raises an exception with a message complaining about the platform:
```
>>> import pathlib
>>> pathlib.WindowsPath("foo.txt")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/alex/.pyenv/versions/3.9.16/lib/python3.9/pathlib.py", line 1084, in __new__
    raise NotImplementedError("cannot instantiate %r on your system"
NotImplementedError: cannot instantiate 'WindowsPath' on your system
>>> 
```

Using `Path()` with a Windows-looking path string produces not the thing we'd hope for. It looks like the whole path is treated as a single file:
```
>>> pathlib.Path("C:\\foo\bar.txt")
PosixPath('C:\\foo\x08ar.txt')
>>> pathlib.Path("C:\\foo\bar.txt").parent
PosixPath('.')
```

In the end, looks like we just need to use `Path()` and run the test suite on different platforms.